packages/hoc/src/deprecated/index.js

```diff
@@ -0,0 +1,21 @@
+// Deprecation HOC care of:
+// https://whawker.github.io/2016/07/15/deprecating-react-components-with-higher-order-components.html
+import React, { Component } from 'react';
+
+const deprecated = (InnerComponent, msg = '') => class extends Component {
+  componentWillMount() {
+    if (process.env.NODE_ENV === 'development') {
```

Comment:
we are distributing this code on npm, we process with babel but not webpack before distributing. Can we be sure that the peer application that is running this code will always have a `process` global? And whether it would contain a key `env`?

I guess React does this somehow for their__DEV__ flags but I've not completely figured it out.

https://github.com/facebook/react/issues/770

Have you already looked in to/considered this? Can we find/link to an established pattern for including dev flags when distributing on npm?

One option, though verbose:

`typeof(process) != 'undefined' && process && process.env && process.env.NODE_ENV === 'development'`

Reply:
A good point well raised.

I had thought about this in passing, but essentially only as far as "Will `process` be there? Isn't that a build-time thing? Oh well it works for me in storybook".

I had considered removing the check completely, and *always* producing a warning in the console, irrespective of the build type.  I'm inclined to think that may be the way to go here.

Reply:
just had a look in a local `node_modules/react/index,js`, and it's this:

https://github.com/facebook/react/blob/b87aabdfe1b7461e7331abb3601d9e6bb27544bc/packages/react/npm/index.js

so thinking if React is doing it then we can do it too.

However I'd match their `=== 'production'` rather than `=== 'development'`

Reply:
i guess in this case it'd be `!== 'production'`

Reply:
⭐️